File: src/main/java/com/hedvig/claims/events/AutomaticPaymentAddedEvent.java
package com.hedvig.claims.events;

import com.hedvig.claims.query.Carrier;
import javax.money.MonetaryAmount;
import lombok.AllArgsConstructor;
import lombok.Value;
import org.axonframework.serialization.Revision;

@Value
@AllArgsConstructor
@Revision("1.0")
public class AutomaticPaymentAddedEvent {
    String Id;
    String claimId;
    String memberId;
    MonetaryAmount amount;
    MonetaryAmount deductible;
    String note;
    boolean isExGracia;
    String handlerReference;
    boolean sanctionCheckSkipped;
    Carrier carrier;
}

File: src/main/java/com/hedvig/claims/events/AutomaticPaymentAddedEvent.kt
package com.hedvig.claims.events

import com.hedvig.claims.query.Carrier
import org.axonframework.serialization.Revision
import javax.money.MonetaryAmount

@Revision("3.0")
class AutomaticPaymentAddedEvent(
    val Id: String,
    val claimId: String,
    val memberId: String,
    val amount: MonetaryAmount,
    val deductible: MonetaryAmount,
    val note: String,
    val isExGracia: Boolean,
    val handlerReference: String,
    val sanctionCheckSkipped: Boolean,
    var carrier: Carrier,
    val payoutDetails : PayoutDetails
)

